sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilterSuite.scala

```diff
@@ -1934,6 +1949,43 @@ abstract class ParquetFilterSuite extends QueryTest with ParquetTest with Shared
       checkAnswer(notIn, Seq())
     }
   }
+
+  private def testStringPredicateWithDictionaryFilter(
+      dataFrame: DataFrame, filter: String): Unit = {
+    Seq(true, false).foreach { enableDictionary =>
+      withTempPath { dir =>
+        val path = dir.getCanonicalPath
+        dataFrame.write
+          .option(ParquetOutputFormat.ENABLE_DICTIONARY, enableDictionary)
+          .parquet(path)
+        Seq(true, false).foreach { pushDown =>
+          withSQLConf(
+            SQLConf.PARQUET_FILTER_PUSHDOWN_STRING_PREDICATE_ENABLED.key -> pushDown.toString) {
+            val accu = new NumRowGroupsAcc
+            sparkContext.register(accu)
+
+            val df = spark.read.parquet(path).filter(filter)
+            df.foreachPartition((it: Iterator[Row]) => it.foreach(v => accu.add(0)))
+            if (enableDictionary && pushDown) {
+              assert(accu.value == 0)
+            } else {
+              assert(accu.value > 0)
+            }
+
+            AccumulatorContext.remove(accu.id)
+          }
+        }
+      }
+    }
+  }
+
+  test("filter pushdown - StringEndsWith/Contains") {
```

Comment:
don't need to test `StringStartsWith` here?

Reply:
`StringStartsWith` push down is an existing feature and has been tested at L1426 https://github.com/apache/spark/blob/5046b8c04cadca6605dd34b98b31850b643dfe45/sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilterSuite.scala#L1426

Reply:
Oh, I mean this test specially does test with `testStringPredicateWithDictionaryFilter`, but I don't see `StringStartsWith` is included here. Don't we need to do `testStringPredicateWithDictionaryFilter` for it?

Reply:
can we merge the existing startswith test into the new one?

Reply:
updated